rex_xai/explanation.py

```diff
@@ -152,90 +204,221 @@ def _explanation(args, model_shape, prediction_func, device):
         avg_box_size,
     )
 
-    exp = Explanation(maps, prediction_func, target, data, args)  # type: ignore
+    run_stats = {
+        'total_passing': total_passing, 
+        'total_failing': total_failing, 
+        'max_depth_reached': max_depth_reached, 
+        'avg_box_size': avg_box_size
+        }
+
+    exp = Explanation(maps, prediction_func, data.target, data, args, run_stats, keep_all_maps=False)
+
+    return exp
+
+
+def analyze(exp: Explanation, data_mode: str, logging_level: int):
+    """Analyzes an Explanation.
 
-    # print(maps)
-    # for k, v in maps.items():
-    #     print(k)
-    #     print(maps.counts)
+    Analyzes the area ratio, entropy difference, insertion and deletion curves for an
+    Explanation object and prints them to ``logging.info``.
 
+    Args:
+        exp: Explanation object as returned by :py:func:`~rex_xai.explanation._explanation`
+        data_mode: Mode of the input data. Entropy difference is only calculated if ``data_mode``
+            is one of ["RGB", "L"].
+        logging_level: used to set logging level back to its original value after ensuring
+            output is printed using :py:func:`logging.info`.
+
+    Returns:
+        None
+
+    """
+    eval = Evaluation(exp)
+    rat = eval.ratio()
+    if data_mode in ("RGB", "L"):
+        be, ae = eval.entropy_loss()  # type: ignore
+        ent = be - ae
+    else:
+        ent = None
+
+    iauc, dauc = eval.insertion_deletion_curve(exp.prediction_func)
+    if logging_level < 2:
```

Comment:
The default level is 0/error. If a user leaves the log level at the default (or intentionally sets it to warning/error), but also uses the `--analyse` flag, they wouldn't see any output without this temporary change in log level! I think it would be possible to do this without realising, and it would be confusing. Would it be better to just use `print()` instead?

I'm also now thinking perhaps it's better to have this function return the analysis results, and handle the printing/logging at a higher level...

Reply:
I think yes on switching to `print()`, if it's something that really should be displayed no matter what the log level is. 

I do like your idea of handling the analysis and printing separately, but perhaps put that in a separate issue as it's nicely self-contained? (Unless you really want to do it now 🙂)